scanner/datastore/postgres/manifest_metadata.go

```diff
@@ -0,0 +1,157 @@
+	}
+
+	return missingManifests, nil
+}
+
+func (i *indexerMetadataStore) StoreManifest(ctx context.Context, manifestID string, expiration time.Time) error {
+	ctx = zlog.ContextWithValues(ctx, "component", "datastore/postgres/indexerMetadataStore.StoreManifest")
+
+	// insertManifest inserts the metadata into manifest_metadata, overwriting the previous expiration, if it exists.
+	const insertManifest = `
+		INSERT INTO manifest_metadata (manifest_id, expiration) VALUES
+			($1, $2)
+		ON CONFLICT (manifest_id) DO UPDATE SET expiration = $2`
+
+	_, err := i.pool.Exec(ctx, insertManifest, manifestID, expiration.UTC())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *indexerMetadataStore) ManifestExists(ctx context.Context, manifestID string) (bool, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "datastore/postgres/indexerMetadataStore.ManifestExists")
+
+	// selectManifest returns 1 if the given manifest exists in the table.
+	// If it does not exist, then no rows will be returned.
+	const selectManifest = `SELECT 1 FROM manifest_metadata WHERE manifest_id = $1`
+
+	row := i.pool.QueryRow(ctx, selectManifest, manifestID)
+	var value int
+	err := row.Scan(&value)
+	if errors.Is(err, pgx.ErrNoRows) {
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("checking if manifest exists: %w", err)
+	}
+
+	// Sanity check the returned value is what is expected.
+	return value == 1, nil
+}
+
+func (i *indexerMetadataStore) GCManifests(ctx context.Context, expiration time.Time) ([]string, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "datastore/postgres/indexerMetadataStore.GCManifests")
+
+	const deleteManifests = `
+		DELETE FROM manifest_metadata
+		WHERE expiration < $1
+		RETURNING manifest_id`
+
+	// Make this a transaction, as failure to delete the manifest should stop the deletion of its metadata.
+	tx, err := i.pool.Begin(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("beginning GCManifests transaction: %w", err)
+	}
+	defer func() {
+		_ = tx.Rollback(ctx)
+	}()
+
+	// Delete expired rows from manifest_metadata
+	rows, err := tx.Query(ctx, deleteManifests, expiration.UTC())
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var deletedManifests []string
+	for rows.Next() {
+		var manifestID string
+		if err := rows.Scan(&manifestID); err != nil {
+			zlog.Warn(ctx).Err(err).Msg("scanning deleted manifest metadata row")
+			continue
+		}
+		deletedManifests = append(deletedManifests, manifestID)
+	}
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("reading deleted manifest metadata rows: %w", err)
+	}
+
+	if i.store != nil {
+		digs := make([]claircore.Digest, 0, len(deletedManifests))
+		for _, m := range deletedManifests {
+			d, err := claircore.ParseDigest(m)
+			if err != nil {
+				return nil, fmt.Errorf("parsing deleted manifest metadata ID: %w", err)
+			}
+			digs = append(digs, d)
+		}
+		deletedDigs, err := i.store.DeleteManifests(ctx, digs...)
```

Comment:
Is there any other metadata related to the IR in CC that we care?

Reply:
This function starts its own transaction for each given `claircore.Digest` and deletes all related data (manifest table, indexreport table, etc). This is what's called when we'd want to call [`libindex.DeleteManifests`](https://github.com/quay/claircore/blob/v1.5.33/libindex/libindex.go#L287)